### PR TITLE
Add tests to verify knife command load & execution

### PR DESCRIPTION
**Pull request by marcparadise**
This verifies that each of the default knife commands will
correctly load by forcing each commnd/subcommand to execute with an
invalid option and confirming that the output shows the command was
loaded.

This will catch future failures like the one fixed in  #11649 

Signed-off-by: Marc A. Paradise <marc.paradise@gmail.com>

### CODE REVIEW

File: knife/spec/support/shared/integration/knife_support.rb [hunk 1]
```diff
@@ -57,14 +57,22 @@ def knife(*args, input: nil, instance_filter: nil)
...
+          instance = subcommand_class.new(args)
+        else
+          # Default behavior doesn't ever set an expectation on Knife::UI which can interfere with caller
+          # expectations when they don't know we're already messing with that.
+          instance = subcommand_class.new(args)
+          instance.ui = Chef::Knife::UI.new(stdout, stderr, stdin, instance.config.merge(disable_editing: true))
+        end
 
         # Load configs
         instance.merge_configs
```
Comment (marcparadise) [inline knife/spec/support/shared/integration/knife_support.rb:74]: I need to make sure this happens before UI setup where we merge config - though locally it did not result in any failures.

File: knife/spec/support/shared/integration/knife_support.rb [hunk 2]
```diff
@@ -79,7 +87,7 @@ def knife(*args, input: nil, instance_filter: nil)
         # smell right, abort.
 
         # To ensure that we don't pick up a user's credentials file we lie through our teeth about
-        # it's existence.
+        # its existence.
```
Comment (lamont-granquist) [inline knife/spec/support/shared/integration/knife_support.rb:90]: i wonder if that line is mine, its the kind of typo that i'd make.

File: knife/spec/support/shared/integration/knife_support.rb (hunk 1 above)
Comment (lamont-granquist) [reply knife/spec/support/shared/integration/knife_support.rb:74]: yeah this is probably fine, its messy, but the whole KnifeSupport thing is inherently pretty messy.

**Review by lamont-granquist — APPROVED** [review]

**Comment by marcparadise** [comment]
Updated this to use bundle exec instead of knife helper; that gives more comfort that we're getting an end-to-end test without shortcutting any of the class loading behaviors.

**Pull request merged**